File: backend/app/schemas/time.py
"""Pydantic schemas for time/geological endpoints."""

from pydantic import BaseModel, Field


class AgeItem(BaseModel):
    age_ma: float
    period: str
    era: str
    epoch: str | None = None
    available_var_count: int = 0


class VariableBrief(BaseModel):
    var_name: str
    display_name_zh: str | None = None
    units: str | None = None


class AgeDetail(AgeItem):
    file_group_id: str
    file_path: str | None = None
    resolution: str | None = None
    available_vars: list[VariableBrief] = []


class TimelineEpoch(BaseModel):
    name: str
    age_range: tuple[float, float]  # [youngest, oldest] Ma
    available_ages: list[float] = []  # data points within this epoch


class TimelinePeriod(BaseModel):
    name: str
    age_range: tuple[float, float]
    epochs: list[TimelineEpoch] = []


class TimelineEra(BaseModel):
    name: str
    periods: list[TimelinePeriod] = []


class TimelineResponse(BaseModel):
    eras: list[TimelineEra] = []

File: frontend/src/types/time.ts
export interface AgeItem {
  age_ma: number
  period: string
  era: string
  epoch: string | null
  available_var_count: number
}

export interface VariableBrief {
  var_name: string
  display_name_zh: string | null
  units: string | null
}

export interface AgeDetail extends AgeItem {
  file_group_id: string
  file_path: string | null
  resolution: string | null
  available_vars: VariableBrief[]
}

export interface TimelineEpoch {
  name: string
  age_range: [number, number]
  available_ages: number[]
}

export interface TimelinePeriod {
  name: string
  age_range: [number, number]
  epochs: TimelineEpoch[]
}

export interface TimelineEra {
  name: string
  periods: TimelinePeriod[]
}

export interface TimelineResponse {
  eras: TimelineEra[]
}
